datasets/flwr_datasets/federated_dataset.py

```diff
@@ -102,9 +113,19 @@ def __init__(
         # Indicate if the dataset is prepared for `load_partition` or `load_full`
         self._dataset_prepared: bool = False
 
-    def load_partition(self, node_id: int, split: Optional[str] = None) -> Dataset:
+    def load_partition(
+        self,
+        node_id: int,
+        split: Optional[str] = None,
+        inner_split: Optional[Union[int, str]] = None,
```

Comment:
That's the part I'm unsure of. It increases the number of parameters => less clarity and the functionality can be easily accomplished by e.g. `load_partition(...)[inner_split]`